conda/api.py

```diff
@@ -1,5 +1,110 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import, division, print_function, unicode_literals
 
-from .core.solve import Solver  # NOQA
-Solver = Solver  # lgtm [py/redundant-assignment]
+from .common.constants import NULL
+from .core.package_cache_data import PackageCacheData as _PackageCacheData
+from .core.prefix_data import PrefixData as _PrefixData
+from .core.solve import Solver as _Solver, DepsModifier as _DepsModifier
+from .core.subdir_data import SubdirData as _SubdirData
+from .models.channel import Channel
+
+
+DepsModifier = _DepsModifier
+
+
+class Solver(object):
+
+    def __init__(self, prefix, channels, subdirs=(), specs_to_add=(), specs_to_remove=()):
+        self._internal = _Solver(prefix, channels, subdirs, specs_to_add, specs_to_remove)
```

Comment:
This will only work for interactive scripting but we could do at least


```python
    self.solve_final_state.__doc__ = self._internal.solve_final_state.__doc__
```

?